### PR TITLE
feat/customer-dashboard page, actions, reducers, saga, api

### PR DESCRIPTION
**Pull request by hemal-digiflux**

### CODE REVIEW

File: src/interfaces/responses/customer-dashboard-response.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface bookingInterface {
+    _id?: string;
```
Comment (Diego-Romero) [inline src/interfaces/responses/customer-dashboard-response.ts]: These fields don't need to be optional

File: src/redux/actions/server/server.actions.ts [hunk 2]
```diff
@@ -35,6 +39,23 @@ export const setCustomerUserAction = (customerUser: CustomerUserResponse): setCu
     customerUser,
 });
 
+export interface fetchCustomerDashboardActionInterface extends Action<'FETCH_CUSTOMER_DASHBOARD'> {}
```
Comment (Diego-Romero) [inline src/redux/actions/server/server.actions.ts]: This should be a type and not an interface, getting a lintin error in my editor.

File: src/redux/reducers/server-reducer/server.reducer.ts [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+
+export interface customerDashboardInterface {
+    upcomingBookings?: bookingInterface[];
+    pastBookings?: bookingInterface[];
 }
 
 export interface ServerStateInterface {
     expertUser: expertUserInterface;
     customerUser?: customerUserInterface;
+    customerDashboard?: customerDashboardInterface;
```
Comment (Diego-Romero) [inline src/redux/reducers/server-reducer/server.reducer.ts]: we need to refactor the stub server as currently we are setting both the customerUserInterface and the customerDashboardInterface inside the customerDashboard Reducer

File: src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx [hunk 4]
```diff
@@ -1,10 +1,152 @@
...
+
+export class CustomerDashboard extends React.Component<CustomerDashboardPageAllProps> {
+    componentDidMount() {
+        this.props.fetchCustomerDashboardAction();
+    }
+
+    renderUpcomingBookings = () => {
+        const rows = [];
+
+        this.props.customerDashboard.upcomingBookings.forEach((booking, index) => {
```
[the remark this answers is not in the document]
Comment (hemal-digiflux) [reply src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx]: Solved!

File: src/interfaces/responses/customer-dashboard-response.ts (hunk 1 above)
Comment (hemal-digiflux) [reply src/interfaces/responses/customer-dashboard-response.ts]: Solved!

File: src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Diego-Romero) [reply src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx]: can't see the changes yet. renderUpcomingBookings and renderPastBookings should be refactored onto a single function that receives bookings I believe.

File: src/redux/reducers/server-reducer/server.reducer.ts (hunk 3 above)
Comment (hemal-digiflux) [reply src/redux/reducers/server-reducer/server.reducer.ts]: Should we make both separate then? Customer Dashboard will have only Appointments data, and CustomerUser will have only UserMe data?

File: src/redux/actions/server/server.actions.ts (hunk 2 above)
Comment (hemal-digiflux) [reply src/redux/actions/server/server.actions.ts]: Please help me with this..

Comment (Diego-Romero) [reply src/redux/actions/server/server.actions.ts]: export **type** fetchCustomerDashboardActionInterface extends Action.....

File: src/redux/reducers/server-reducer/server.reducer.ts (hunk 3 above)
Comment (Diego-Romero) [reply src/redux/reducers/server-reducer/server.reducer.ts]: yes indeed

File: src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (hemal-digiflux) [reply src/ecosystems/CustomerDashboard/CustomerDashboard.page.tsx]: created a common method => renderBookings()

**Pull request merged**